### DISCUSSION TITLE
Potential move of Aries to the OWF

### DISCUSSION THREAD

**Discussion by TelegramSam** [hyperledger/aries#20] [General]
On our Aries WG call yesterday, our discussion on this topic yielded the following notes. Please add discussion here in advance of our meeting next week.

- High ambition level of OWF
	- Goal is to create code that has relevance in code that meets regulatory requirements (e.g. as part of the EU Digital Identity Wallet)
	- Aries features and approaches can gain better representation in compliance and regulatory settings employing wallets (key leverage point for Aries community)
	- Outreach to European Commission already established at OWF
- Funding
	- 300 orgs listed in doc
	- https://openwallet.foundation/wp-content/uploads/sites/11/2023/02/OpenWallet-Foundation-Overview.pdf
	- 41 listed current sponsors
	- 7 million draft budget?
	- Funding state? Will fundraising match needs?
	- Difficulty of orgs being members of All The Things.
		- Time
		- Money
- Project Support from Hyperledger
	- Hyperledger social reach is larger
	- Training workshop organized by HyperLedger
- Code and Standards being co-located in the same org is difficult
- Aries implementing standards designed outside of 
	- Aries
	- mDL
	- OpenID4VerifiableCredentials
	- Beyond Aries?
- OWF explicitly does not want to create standards, OWF core is creating code to implement standards
- Perception of the scope of Aries? What Aries does/is:
	- RfCs
	- Code matching RfCs
	- Interop Profiles
- ToDo: More clearly communicate what Aries does, also towards OWF

**Comment by TimoGlastra** [comment]
# Thoughts on moving Hyperledger Aries to the Open Wallet Foundation

There’s been a lot of discussion ongoing in the Hyperledger Aries project lately about the potential move of the project to the Open Wallet Foundation. I’d like to leave some comments on the matter. These comments are my personal opinion, but are also based on a lot of different things I’ve heard within my team at Animo, in discussions with other people involved in the digital identity space, and other members of the Aries and Aries Framework JavaScript communities. 

**Goal of the overarching foundations**

Hyperledger is “The open global ecosystem for enterprise grade blockchain technologies”. The focus of the Hyperledger projected is rooted in blockchain and ledger technology. 

The goal of Aries has a wider scope than just blockchain, as it focuses on the use of verifiable credentials:

> Hyperledger Aries provides a shared, reusable, interoperable tool kit designed for initiatives and solutions focused on creating, transmitting and storing verifiable digital credentials. It is infrastructure for blockchain-rooted, peer-to-peer interactions. This project consumes the cryptographic support provided by Hyperledger Ursa, to provide secure secret management and decentralized key management functionality.


but still defines itself as “… infrastructure for blockchain-rooted, peer-to-peer interactions.”, so blockchain is a vital part of what Aries is.

If we look at the mission of the open wallet foundation, I think it aligns much more with what Aries intends to be:

> The OWF is a consortium of companies and non-profit organisations collaborating to drive global adoption of open, secure and interoperable digital wallet solutions as well as providing access to expertise and advice through our Government Advisory Council.
>
> The OWF aims to set best practices for digital wallet technology through collaboration on standards-based OSS components that issuers, wallet providers and relying parties can use to bootstrap implementations that preserve user choice, security and privacy.

In my opinion the name Open Wallet Foundation is terrible, as wallets are an overused term from the crypto community, it’s ambiguous what a wallet exactly is, etc.. If we were to replace every occurrence of the word **wallet** with **agent,** I think we would have the perfect mission for what Aries is. As Alex Andrei mentioned, “Where’s the Open Agent Foundation?”.

**Misconception about what Aries is**

I think this point is very related to the previous point about the goal and scope of the Aries project and the Hyperledger Foundation. 

I think the most raised point I’ve heard in this discussion is the misconception about what Aries is. People see Aries as agents that implement DIDComm to exchange AnonCreds credentials anchored on an Hyperledger Indy network. 


The post from Kaliya on [Being “Real” about Hyperledger Indy & Aries / Anoncreds](https://identitywoman.net/being-real-about-hyperledger-indy-aries-anoncreds/) is the perfect example of this. It’s so full of misconceptions about what Aries is and that Aries can only be used with AnonCreds and Hyperledger Indy. But I don’t blame her. We’ve done a terrible job at showing the contrary. Most deployments of Hyperledger Aries today are Hyperledger Indy, DIDComm v1 and AnonCreds based (Aries Framework Go has been an exception on this matter).


Another great example was shared today by Tim Bloomfield. The following quote is from a paper form the FIDO alliance : “Hyperledger Indy SDK Wallet [74] is a cloud-based wallet, which is accessed by the Hyperledger Aries protocols. This cloud-based architecture is most relevant for the EUDI Wallet Form Factor 2.” (link to* [paper](https://media.fidoalliance.org/wp-content/uploads/2023/04/FIDO-EUDI-Wallet-White-Paper-FINAL.pdf))


To show the contrary: At Animo we’re currently working on a mobile wallet built on Aries Framework JavaScript, that is not connected to any blockchain, does not support Hyperledger Indy, is not connected to any blockchain, has no support for DIDComm or AnonCreds, does not use the Indy SDK, and primarily supports OpenID4VCI, SIOPv2 and OpenID4VP with W3C JWT Verifiable Credentials. I think most people won’t know that it is even a possibility to do this with an Aries agent.


Sam Curren mentioned the continuity of brand being an important factor in a potential move to the OWF. I would say the opposite is true. The biggest thing we would have to gain is a new brand and identity for the immensely advanced and powerful agents that are being developed under the Aries projects currently.


So separate of whether Aries is moved to OWF, we should:

* Freshen up the Aries brand, get rid of outdated code and documents that conflict with the current mental model of Aries
  * Archive outdated repositories
  * Clean up the documentation, descriptions of Aries. 
  * Repurpose the goal of Aries. It should not be tied to blockchain at all
* Carefully look up all references where Indy, the Indy SDK, and AnonCreds are used as always being used in Hyperledger Aries. We really need to get rid of the confusion.
*  Really put out the message what Aries is and reiterate the scope using e.g. blog posts, video tutorials, whatever.


**Colocation of code and standards**

Hyperledger and Aries are focused on implementation rather than standards. This means standards are implementation driven, and different implementations often implement based on other implementations. But Aries is also a place to standardize through the use of RFCs, and reach a common profile for interoperability through the Aries Interop Profile (AIP).


Having the specifications in the same place as the code means there’s more confusion about what Aries is and what it support. There’s a big difference in what Aries can support in terms of AIP, or what it can support in terms of what is actually implemented. A common practice we see across the industry is that a lot of the foundations concern about standards, not implementations. It’s not that implementations are not important, but I do think there’s a good reason to keep them separate.


And now that most of the standards have already been extracted from Indy/Aries (AnonCreds, DIDComm), what is Aries really besides it’s agent implementations? There’s a full DIDComm based interop profile under DIF ([WACI DIDComm](https://identity.foundation/waci-didcomm/)), which does point to the Aries RFCs, but only for protocol definitions. Those RFCs could be moved to [didcomm.org](https://didcomm.org/). 


The Open Wallet Foundation is very clear in that it doesn’t want to create standards, just open source code, which I very much like.


So separate from whether Aries is moved to the OWF, I think the following would make sense:

* The Aries RFCs are deprecated and moved to didcomm.org as much as possible
* DIF could host the Aries Interop Profile (or “Something Interop Profile”), that is agnostic in the technologies and standards it ***can*** use (it should not be a companion to DIDComm)


In the case Aries stays a Hyperledger project, Aries is now fully focused on Agent implementations, not standards. In the case Aries moves to Open Wallet Foundation, it is also solely focused on Agent implementations, not standards.


**Support from Hyperledger**

This is also a big topic I’ve heard a lot of times. The support we get from Hyperledger is fantastic. There’s really nothing to say about that. We get support from a dedicated team, Github Runners, Infrastructure, a marketing team, workshops, and a great foundation for hosting a community, a big following, and a lot of paying members.


I don’t see Hyperledger going anywhere in the coming years, and I’m not so sure about the support the Open Wallet Foundation can provide yet. There’s ongoing discussion about what that will look like, and I’m interested to hear more about what the Open Wallet Foundation has to offer our community. In addition, I’ve heard some semi concerning things about how funds are allocated, which I don’t know the slightest thing about, but is something I’d like to learn more about. I’m still undecided whether OWF really cares about Aries as a project, or whether it’s a marketing play and show how much momentum they got (there’s been pressure on making a decision quickly).

---

With all points being raised I see potential in moving Aries to OWF, but I also see a lot of potential in keeping Aries a Hyperledger project. I think the most important thing to do is really define what Aries is about, and share that with the world. What is Aries? According to John, and which I fully agree with: “A set of frameworks that can work with protocols, credential types, crypto, etc. from a wide variety of sources”. That purpose could be fulfilled in both organizations. Most of the problems I’ve outlined above are issues within the project itself, and stem from a lack of effort (we at animo play a big role in this as well) put in sending out the message (and showing, not telling) that Aries is not DIDComm, AnonCreds & Indy. 

If the OWF can provide a similar level of support as Hyperledger does, and really sees the values of what we’ve built over the past few years, I’m a proponent of a move to the Open Wallet Foundation.

**Reply by TelegramSam** [reply]
This is the best summary and clearest set of proposals we have seen to date. Thank you Timo.

>In my opinion the name Open Wallet Foundation is terrible, as wallets are an overused term from the crypto community, it’s ambiguous what a wallet exactly is, etc.. If we were to replace every occurrence of the word wallet with agent, I think we would have the perfect mission for what Aries is. As Alex Andrei mentioned, “Where’s the Open Agent Foundation?”.

Well stated. I think this division, the difference between a wallet focused on keys and credentials and an agent focused on connections and relationships (that happen to use keys and verifiable credentials) is actually a huge largely unseen issue in the DI/SSI space. I thnk more discussion on this topic is warranted (and on the schedule for next Wed's Aries call).

>Sam Curren mentioned the continuity of brand being an important factor in a potential move to the OWF. I would say the opposite is true. The biggest thing we would have to gain is a new brand and identity for the immensely advanced and powerful agents that are being developed under the Aries projects currently.

This has me thinking. I think the part I'd wish to retain is the long experience we have as a community. I'd love to shed the misconceptions. Maybe getting both isn't possible.

>If the OWF can provide a similar level of support as Hyperledger does, and really sees the values of what we’ve built over the past few years, I’m a proponent of a move to the Open Wallet Foundation.

I agree with the general sentiment. Evaluating the level of support is quantifiable. Seeing the value of what we've built is really squishy. Perhaps this relates more to the wallet vs agent discussion, and the level of alignment that OWF has with Agents.

**Reply by darrellodonnell** [reply]
I like your thinking here. 

I believe that the AIP (Aries Interop Profile) test harness and test suites need to be separated in some way as well. Some of the guts belong, but particular profiles should live with an ecosystem/capability-specific area. Without the separation of the interop suites, Aries will always be painted as self-dealing when we talk about interop.

**Reply by talltree** [reply]
> I think this division, the difference between a wallet focused on keys and credentials and an agent focused on connections and relationships (that happen to use keys and verifiable credentials) is actually a huge largely unseen issue in the DI/SSI space. I thnk more discussion on this topic is warranted (and on the schedule for next Wed's Aries call).

+1000 to this point. In the world of SSI, wallets and agents are inextricably interconnected — like a hard drive and an operating system. But  90% of the time the term "wallet" is being used to describe both. IMHO that's completely backwards—over time it's the agent that will provide the lion's share of value.

I currently have no idea what to do about this situation; I just wanted to highlight that: a) I deeply agree, and b) this semantic issue (and marketing issue) affects Aries and OpenWallet equally — so one way or another we should collaborate on solving it.

**Reply by kurtmilam** [reply]
I'll chime in as someone leading a team implementing a pilot including agents that will issue, hold and verify credentials for an EU national government. We are currently working with AFJ 0.4.0 and are hoping that support for other credential and protocol standards will be available in time to switch from pilot to production.

> wallets are an overused term from the crypto community, it’s ambiguous what a wallet exactly is, etc.. If we were to replace every occurrence of the word wallet with agent, I think we would have the perfect mission for what Aries is. As Alex Andrei mentioned, “Where’s the Open Agent Foundation?”.

100%. Our team has mostly switched from talking about 'wallets' to talking about 'agents'. It took a little time to understand that any agent can be any combination of Issuer, Verifier and/or Holder in different scenarios.

> People see Aries as agents that implement DIDComm to exchange AnonCreds credentials anchored on an Hyperledger Indy network.

100%. We get a lot of pushback when we mention that we're working with Aries. Whether from competing (commercial) frameworks or other folks in the ecosystem who should be Aries supporters.

For instance, I was told by someone in the ecosystem that  (summarized), "Aries is outdated. It's mostly used for legacy projects, and most projects in the EU have moved on from it." This was said to me by someone who should be an Aries supporter (in the sense that they want the ecosystem to thrive, don't have a reason to back a specific framework other than standards-conformance, and they should be supportive of anyone who is contributing to the ecosystem).

From competing (commercial) frameworks, the pushback is even harder, along the lines of, "Aries is the wrong choice, full stop. They don't support the right standards (wrt eIDAS, ARF, EUDI Wallet conformance) and probably never will. Anyone who recommends them doesn't know what they're talking about."
